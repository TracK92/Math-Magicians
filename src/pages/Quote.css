* {
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
}

.quote {
  width: 90%;
  height: 30vh;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: auto;
  font-weight: bold;
  font-size: 1.5rem;
}

.quotes_img {
  height: 250px;
  display: flex;
  justify-content: center;
  margin: auto;
  align-items: center;
}

.quotes_div {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

@media screen and (max-width: 1050px) {
  .quote {
    font-size: 1.2rem;
  }
}
