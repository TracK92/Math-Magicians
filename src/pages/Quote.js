import React, { useEffect } from 'react';
import './Quote.css';
import logo from '../assets/4.png';

const Quote = () => {
  // create an array of mathematics quotes

  const quotes = [
    {
      id: 1,
      quote: 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding –William Paul Thurston',
    },
    {
      id: 2,
      quote: 'Mathematics is the key and door to the sciences –Galileo Galilei',
    },
    {
      id: 3,
      quote: 'Mathematics is the music of reason –James Joseph Sylvester',
    },
    {
      id: 4,
      quote: 'Wherever there is number, there is beauty –Procus',
    },
    {
      id: 5,
      quote: 'The study of mathematics, like the Nile, begins in minuteness but ends in magnificence –Charles Caleb Colton',
    },
    {
      id: 6,
      quote: 'Life is a math equation. In order to gain the most, you have to know how to convert negatives into positives -Anonymous',
    },
    {
      id: 7,
      quote: 'We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics –Catherine Johnson, African-American mathematician',
    },
    {
      id: 8,
      quote: 'Mathematics is the most beautiful and most powerful creation of the human spirit –Stefan Banach',
    },
    {
      id: 9,
      quote: 'Pure Mathematics is, in its way, the poetry of logical ideas –Albert Einstein',
    },
    {
      id: 10,
      quote: 'Mathematics has beauty and romance. It’s not a boring place to be, the mathematical world. It’s an extraordinary place; it’s worth spending time there - Marcus du Sautoy',
    },
  ];

  // reload page every 10 seconds
  useEffect(() => {
    const interval = setInterval(() => {
      window.location.reload();
    }, 5000);
    return () => clearInterval(interval);
  }, [quotes]);

  return (
    <div className="quotes_div">
      <img src={logo} alt="logo" className="quotes_img" />
      <p className="quote" style={{ textAlign: 'center' }}>
        {quotes[Math.floor(Math.random() * quotes.length)].quote}
      </p>
    </div>
  );
};

export default Quote;
