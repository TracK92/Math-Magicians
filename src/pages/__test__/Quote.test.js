import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Quote from '../Quote';

test('should render the Quote page', () => {
  const tree = renderer.create(<Router><Quote /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
