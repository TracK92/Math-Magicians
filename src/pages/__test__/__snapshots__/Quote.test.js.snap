// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Quote page 1`] = `
<div>
  <p
    className="quote"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Mathematics has beauty and romance. It’s not a boring place to be, the mathematical world. It’s an extraordinary place; it’s worth spending time there - Marcus du Sautoy
  </p>
</div>
`;
